src/Qwiq.Core/IQueryFolderCollection.cs

```diff
@@ -0,0 +1,8 @@
+using System;
+
+namespace Microsoft.Qwiq
+{
+    public interface IQueryFolderCollection : IReadOnlyObjectWithIdCollection<IQueryFolder, Guid>, IEquatable<IQueryFolderCollection>
```

Comment:
Do we need to expose new types here over just `IReadOnlyCollection<T>`?

Reply:
Yes to support look ups by Name and Id.

Reply:
Indexing vs needing to use Single/First